pkg/diagnostics/collectors.go

```diff
@@ -380,6 +380,56 @@ func (c *collectorFactory) crdCollector(crdType string) *Collect {
 	}
 }
 
+// APIServerCollector collect connection info when running a pod on an existing cluster
+func (c *collectorFactory) APIServerCollector(controlPlaneIP string) []*Collect {
+	var collectors []*Collect
+	collectors = append(collectors, c.generateRunPodsCollector(controlPlaneIP)...)
+	// collectors = append(collectors, c.pingLogCollectors()...)
+	return collectors
+}
+
+func (c *collectorFactory) generateRunPodsCollector(controlPlaneIP string) []*Collect {
+	return c.runPodsCollectors(controlPlaneIP)
+}
+
+func (c *collectorFactory) runPodsCollectors(controlPlaneIP string) []*Collect {
+	tempIPRequest := "yum install -y nmap && for port in 6443 22; do nc -z -v -w5 " + controlPlaneIP + " $port; done"
```

Comment:
Why not use a Golang net library instead of exec'ing nc? Here's an example of how we use it currently https://github.com/aws/eks-anywhere/blob/main/pkg/networkutils/networkutils.go#L32-L44

Reply:
this is running in the container in the pod

Reply:
So is the container going to just run these commands directly through a shell script or something?
Or is there a custom go binary that it's going to execute?
I was thinking it was the latter but if it isn't then ignore my comment